src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -905,23 +932,25 @@ private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(MethodDeclaration
             return result.ToImmutableAndFree();
         }
 
+        const DeclarationModifiers PartialMethodExtendedModifierMask =
+            DeclarationModifiers.Virtual |
+            DeclarationModifiers.Override |
+            DeclarationModifiers.New |
+            DeclarationModifiers.Sealed |
+            DeclarationModifiers.Extern;
+
+        internal bool HasExtendedPartialModifier => (DeclarationModifiers & PartialMethodExtendedModifierMask) != 0;
+
         private void CheckModifiers(bool isExplicitInterfaceImplementation, bool hasBody, Location location, DiagnosticBag diagnostics)
         {
-            const DeclarationModifiers partialMethodInvalidModifierMask = (DeclarationModifiers.AccessibilityMask & ~DeclarationModifiers.Private) |
-                     DeclarationModifiers.Virtual |
-                     DeclarationModifiers.Abstract |
-                     DeclarationModifiers.Override |
-                     DeclarationModifiers.New |
-                     DeclarationModifiers.Sealed |
-                     DeclarationModifiers.Extern;
-
             bool isExplicitInterfaceImplementationInInterface = isExplicitInterfaceImplementation && ContainingType.IsInterface;
 
-            if (IsPartial && !ReturnsVoid)
+            if (IsPartial && (!ReturnsVoid || HasExplicitAccessMod || HasExtendedPartialModifier))
             {
-                diagnostics.Add(ErrorCode.ERR_PartialMethodMustReturnVoid, location);
```

Comment:
>ERR_PartialMethodMustReturnVoid [](start = 42, length = 31)

If we are no longer reporting this error, please consider commenting out the `ErrorCode` field and removing the resource.

Reply:
Done.